tests/ui/check-cfg/my-awesome-platform.json

```diff
@@ -8,5 +8,6 @@
     "os": "ericos",
     "linker-flavor": "ld.lld",
     "linker": "rust-lld",
+    "relocation-model": "static",
```

Comment:
I assume this will cause a bunch of breakage for custom target specs out there as well... the default `relocation_model` is "pic", but the default for `dynamic_linking` is `false`, and that's not a coherent combination (or at least, it is rejected by our existing target checks). Therefore, all targets *must* explicitly set one of those two fields.

Reply:
Yeah, most of our tests violate this.

So either it's actually not so bad to have "relocation_model: pic, dynamic_linking: false", or we should really use different defaults...

Reply:
We need some experts for linking and target specs... Cc @bjorn3 , not sure who else?

Reply:
I've made it so that JSON targets may violate this particular check. Given that we also allow uefi targets to violate it, I assume this invariant is not too critical.

Reply:
The combination of relocation model set to PIC and no dynamic linking is definitively fine when `static_position_independent_executables` is enabled. In fact `static_position_independent_executables` requires either PIC or PIE. When `static_position_independent_executables` is false, I think it would also be fine as the linker can just fill in a fixed address in the GOT, but I'm not entirely sure linkers actually support that.

Reply:
`static_position_independent_executables` implies `position_independent_executables` and that is enough to pass this check. However, `static_position_independent_executables`  also defaults to false.